pyanaconda/modules/network/nm_client.py

```diff
@@ -1217,3 +1217,75 @@ def get_slaves_from_connections(nm_client, slave_type, master_specs):
             if iface:
                 slaves.add((iface, con.get_uuid()))
     return slaves
+
+
+def get_config_file_connection_of_device(nm_client, device_name, device_hwaddr=None):
+    """Find connection of the device's configuration file.
+
+    :param nm_client: instance of NetworkManager client
+    :type nm_client: NM.Client
+    :param device_name: name of the device
+    :type device_name: str
+    :param device_hwaddr: hardware address of the device
+    :type device_hwaddr: str
+    :returns: uuid of NetworkManager connection
+    :rtype: str
+    """
+
+    cons = []
+    for con in nm_client.get_connections():
+
+        filename = con.get_filename() or ""
+        # Ignore connections from initramfs in
+        # /run/NetworkManager/system-connections
+        if filename.startswith("/run"):
+            continue
+        con_type = con.get_connection_type()
+
+        if con_type == '802-3-ethernet':
+
+            # Ignore slaves
+            if con.get_setting_connection().get_master():
+                continue
+
+            interface_name = con.get_interface_name()
+            mac_address = None
+            wired_setting = con.get_setting_wired()
+            if wired_setting:
+                mac_address = wired_setting.get_mac_address()
+
+            if interface_name:
+                if interface_name == device_name:
+                    cons.append(con)
+            elif mac_address:
+                if device_hwaddr:
+                    if device_hwaddr.upper() == mac_address.upper():
```

Comment:
(Optional): I would start thinking if we shouldn't have simple object to store MAC address, IP address and similar and to have comparation of two classes instead of having ``.upper()`` and similar code on all the places. It would be pretty easy to test these.